src/main/java/edu/kis/powp/jobs2d/drivers/adapter/controller/DeviceControllerDriver.java

```diff
@@ -2,52 +2,46 @@
 
 import edu.kis.powp.jobs2d.drivers.VisitableJob2dDriver;
 import edu.kis.powp.jobs2d.drivers.adapter.DriverDecorator;
-import edu.kis.powp.jobs2d.drivers.singleton.DeviceController;
+import edu.kis.powp.jobs2d.drivers.singleton.DeviceManager;
 
-import java.util.ArrayList;
-import java.util.List;
 
 public class DeviceControllerDriver implements DriverDecorator {
 
     private VisitableJob2dDriver driver;
-    private final List<IThresholdListener> listeners = new ArrayList<>();
-    private double distanceThreshold = 0;
+
 
     public DeviceControllerDriver() {
-        addListener(new LoggingThresholdListener());
+        DeviceManager.getInstance().addThresholdListener(new LoggingThresholdListener());
```

Comment:
SRP - listeners should be configured elsewhere. They are still configured inside DeviceControllerDriver!
addThresholdListener can be easily called from outside this class.